### ISSUE TITLE
The classification metric of accuracy, precision, and recall differs from scikit-learn

### ISSUE THREAD

**Issue by nathesh** [IBM/AIF360#73]
```
import pandas as pd
import sys
import numpy as np
np.random.seed(0)
from aif360.datasets import StructuredDataset as SD
from aif360.datasets import BinaryLabelDataset as BLD
from aif360.metrics import ClassificationMetric as CM
from aif360.metrics import BinaryLabelDatasetMetric
from aif360.algorithms.preprocessing import Reweighing
from sklearn.ensemble import RandomForestClassifier as RF
from sklearn.datasets import make_classification as mc 
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score

data, label = mc(n_samples=10000,n_features=30)
bias_feature = label.copy()
np.random.shuffle(bias_feature)
agg_data = np.hstack([data,  bias_feature.reshape(-1,1), label.reshape(-1,1),])
pd_data = pd.DataFrame(agg_data, columns=list(range(1,31)) + ["gender", "labels"])
dataset = BLD(favorable_label=0, unfavorable_label=1,df=pd_data,
              label_names=["labels"], protected_attribute_names=["gender"], 
              privileged_protected_attributes=[2])
dataset_orig_train, dataset_orig_test = dataset.split([0.7], shuffle=True)
dataset_orig_test_pred = dataset_orig_test.copy(deepcopy=True)
privileged_groups = [{'gender': 0}]
unprivileged_groups = [{'gender': 1}]
metric_orig_train = BinaryLabelDatasetMetric(dataset_orig_train, 
                                             unprivileged_groups=unprivileged_groups,
                                             privileged_groups=privileged_groups)
clf = RF()
clf.fit(dataset_orig_train.features,dataset_orig_train.labels)

predictions = clf.predict(dataset_orig_test.features)
proba_predictions = clf.predict_proba(dataset_orig_test.features)

dataset_orig_test_pred.scores = proba_predictions[:,0].reshape(-1,1)
dataset_orig_test_pred.labels = predictions.reshape(-1, 1)

cm_pred_valid = CM(dataset_orig_test, dataset_orig_test_pred, unprivileged_groups=unprivileged_groups,
                             privileged_groups=privileged_groups)

cm = ["precision","recall", "accuracy"]


metrics = {}
for c in cm:
    metric = eval("cm_pred_valid." + c + "()")
    metrics[c] =  metric


metrics["recall"], metrics["accuracy"], metrics["precision"]


print("Scikit-learn metrics")
for key,value in {"recall": recall_score,"accuracy": accuracy_score, "precision": precision_score}.items():
    metric = value(dataset_orig_test.labels,predictions)
    print("{} score is: {}".format(key,metric))

print("AIF360 metrics")
for key in ["recall","accuracy", "precision"]:
    print("{} score is: {}".format(key,metrics[key]))
``` 

produces the following:

i.e. for scikit-learn
```
recall score is: 0.8780649436713055
accuracy score is: 0.8856666666666667
precision score is: 0.8928571428571429
```
and for AIF360

```
recall score is: 0.8933601609657947
accuracy score is: 0.8856666666666667
precision score is: 0.8786279683377308
```

**Comment by nathesh** [comment]
The issue is also being tracked --> https://github.com/aif360-learn/aif360-learn/issues/43

**Comment by hoffmansc** [comment]
The issue was you initialized the BLD with `favorable_label=0` but sklearn does `pos_label=1` by default. To solve this you could add `pos_label=0` to `recall_score` and `precision_score` or change to `favorable_label=1`, depending on your application
